src/attackmate/executors/vnc/vncexecutor.py

```diff
@@ -0,0 +1,154 @@
+from attackmate.processmanager import ProcessManager
+from attackmate.schemas.vnc import VncCommand
+from vncdotool import api
+from vncdotool.client import AuthenticationError
+from attackmate.executors.vnc.sessionstore import SessionStore
+from attackmate.executors.executor_factory import executor_factory
+import time
+
+@executor_factory.register_executor('vnc')
+class VncExecutor(BaseExecutor):
+    def __init__(self, pm: ProcessManager, cmdconfig=None, *, varstore: VariableStore):
+        self.session_store = SessionStore()
+        self.set_defaults()
+        super().__init__(pm, varstore, cmdconfig)
+
+    def set_defaults(self):
+        self.hostname = None
+        self.port = None
+        self.display = None
+        self.password = None
+
+
+    def build_connection_string(self) -> str:
+        if not self.hostname:
+            raise ExecException("Hostname is required for VNC connection.")
+        connection_str = self.hostname
+        if self.display is not None:
+            connection_str += f":{self.display}"
+        elif self.port is not None:
+            connection_str += f"::{self.port}"
+        return connection_str
+
+    def connect(self, command: VncCommand) -> api.ThreadedVNCClientProxy:
+        
+
+        client = api.connect(self.build_connection_string(), command.password)
+        time.sleep(5)
+        if not (client and client.protocol and client.protocol.connected):  
+            self.logger.info(f"Could not connect to VNC server: {self.build_connection_string()}")
+            client.disconnect()
+            return None
+        else:
+            return client
+
+    def connect_use_session(self, command):
+
+        if command.creates_session is not None:
+            # If 'creates_session' is specified, create a new session and save it
+            client = self.connect(command)
+            if not client:
+                return None
+            self.session_store.set_session(command.creates_session, client)
+            return client
+
+        if command.session is not None:
+            # If 'session' is specified, check if it exists, else raise an error
+            if not self.session_store.has_session(command.session):
+                raise ExecException(f'VNC-Session not in Session-Store: {command.session}')
+            else:
+                return self.session_store.get_client_by_session(command.session)
+
+        # If neither 'creates_session' nor 'session' is provided
+        default_session = 'default'
+
+        if not self.session_store.has_session(default_session):
+            # Create default session if it doesn't exist
+            client = self.connect(command)
+            self.session_store.set_session(default_session, client)
+        else:
+            # Retrieve the default session if it already exists
+            client = self.session_store.get_client_by_session(default_session)
+
+        return client
+
+    def cache_settings(self, command: VncCommand):
```

Comment:
I think we don't need the cache for the vnc-executor. (I even believe that we might remove it at some point from the sshexecutor too)

Reply:
removed